crates/runtime/src/component/view.rs

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use datafusion::sql::TableReference;
+use snafu::prelude::*;
+use spicepod::component::view as spicepod_view;
+use std::fs;
+
+use super::dataset::Dataset;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Unable to load SQL file {file}: {source}"))]
+    UnableToLoadSqlFile {
+        file: String,
+        source: std::io::Error,
+    },
+
+    #[snafu(display(
+        "The view is uninitialized, please report a bug at https://github.com/spiceai/spiceai"
+    ))]
+    Uninitialized,
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct View {
+    pub name: TableReference,
+    /// Inline SQL that describes a view.
+    sql: Option<String>,
+    /// Reference to a SQL file that describes a view.
+    sql_ref: Option<String>,
```

Comment:
To make it clear you cannot have both, you could have a single String and a flag field for the type of String.

Reply:
I reworked this so that we fail during the spicepod conversion if they don't specify one.